### PR TITLE
openvdb: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

**Comment by valgur** [comment]
TODO: add `bison` and `flex` deps for `ax`.

**Comment by perseoGI** [comment]
@valgur why have you removed v8.0.1?

**Comment by valgur** [comment]
@perseoGI If I recall correctly, it's CMake parameters were considerably different from newer versions and it did not seem worthwhile to keep supporting it. Do you need v8, specifically?

**Review by jwillikers — APPROVED** [review]

File: recipes/openvdb/all/conanfile.py [hunk 1]
```diff
@@ -16,232 +21,283 @@ class OpenVDBConan(ConanFile):
...
+            "Disabled by default in ConanCenter to avoid excessive memory usage during compilation."
+        ),
+        "use_imath_half": (
+            "Use the definition of half-precision floating point types from the Imath library. "
+            "If False, the embedded definition provided by OpenVDB is used. "
+            "You may set this to on to force Imath half to be used if you know it to be required."
+        ),
+        "with_blosc": "Use Blosc for improved disk compression. Recommended.",
+        "with_log4cplus": "Use log4cplus for improved OpenVDB Logging.",
+        "with_zlib": "Use ZLib for disk serialization compression. ZLib can only be disabled if Blosc is also disabled.",
```
Comment (RubenRBS) [inline recipes/openvdb/all/conanfile.py:80]: Thanks for adding these lately, it's super useful for recipe consumers :)

File: recipes/openvdb/all/conanfile.py [hunk 2]
```diff
@@ -16,232 +21,283 @@ class OpenVDBConan(ConanFile):
...
             self.requires("zlib/[>=1.2.11 <2]")
-        if self.options.with_exr:
-            # Not necessary now. Required for IlmBase::IlmImf
-            self.requires("openexr/2.5.7")
         if self.options.with_blosc:
-            self.requires("c-blosc/1.21.1")
+            self.requires("c-blosc/1.21.5")
         if self.options.with_log4cplus:
-            self.requires("log4cplus/2.0.7")
+            # TODO: OpenVDB requires log4cplus v1.*, which is not available in CCI
```
Comment (RubenRBS) [inline recipes/openvdb/all/conanfile.py:142]: What does this one mean? 1.2.2 does exist, so what's the comment referring to?

**Review by RubenRBS — COMMENTED** [review]
Thanks! Sorry for the delay on getting this reviewed, looks good, I only have a few questions :)

**Review by franramirez688 — COMMENTED** [review]
It looks great! Some minor comments

File: recipes/openvdb/all/conanfile.py (hunk 2 above)
Comment (valgur) [reply recipes/openvdb/all/conanfile.py:142]: Yeah, I added the v1.2.2 in a later PR (#21904). :)
I'll update the recipe.

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**